web/src/app/converters/firebase-data-converter.ts

```diff
@@ -465,10 +465,32 @@ export class FirebaseDataConverter {
     );
   }
 
+  private static toResults(job: Job, data: DocumentData): Map<string, Result> {
+    const submissionData = data.data ?? data.results ?? data.responses;
+    return Map<string, Result>(
+      keys(submissionData)
+        .map((taskId: string) => [
+          taskId as string,
+          FirebaseDataConverter.toResult(
+            job.tasks!.get(taskId)!,
```

Comment:
We should also probably handle this case explicitly, and/or comment describing why this can never happen. I'd also suggest adding a brief comment to TSDoc that this method handles errors, i.e., caller doesn't need to catch.

Alternatively, we can have this method return `Map<String, Result | Error>` and let the caller filter and log.

Reply:
Added TSDoc mentioning that this method handles errors. What do you mean by `this can never happen`?

Reply:
Thanks! Re "this", I was referring to job.tasks being `null` or `undefined`. It can never happen if we checked it previously. If that's not the case, we should also handle this case.

Reply:
Got it, we did check `job.tasks` before this. I changed the TSDoc regarding this and changed the null/undefined check a bit.